botorch/utils/transforms.py

```diff
@@ -166,7 +171,25 @@ def decorated(cls: Any, X: Tensor, **kwargs: Any) -> Any:
                     f" got X with shape {X.shape}."
                 )
             X = X if X.dim() > 2 else X.unsqueeze(0)
-            return method(cls, X, **kwargs)
+            output = method(cls, X, *args, **kwargs)
+            if (
```

Comment:
This whole if clause is kind of complex, mind putting this into a helper function that returns a bool? That will clean things up. 

Also, can you migrate this from using the `_input_batch_shape` to the universal `batch_shape` interface introduced in #588?